homeassistant/components/goodwe/services.yaml

```diff
@@ -0,0 +1,33 @@
+set_work_mode:
+  description: |
+    Set the inverter work_mode.
+    (Use with caution, it modifies inverter's operational parameter, usually accessible to installers only.)
+  fields:
+    entity_id:
+      description: ID of the inverter entity
+      example: "sensor.goodwe_inverter"
+    work_mode:
+      description: 0 - General mode, 1 - Off grid mode, 2 - Backup mode, 3 - Wait mode
+      example: "0"
+set_ongrid_battery_dod:
+  description: |
+    Set the battery depth of discharge (on-grid)
+    (Use with caution, it modifies inverter's operational parameter, usually accessible to installers only.)
+  fields:
+    entity_id:
+      description: ID of the inverter entity
+      example: "sensor.goodwe_inverter"
+    ongrid_battery_dod:
+      description: 0 - 89, Depth of Discharge %
```

Comment:
I'd make this a number entity.

Reply:
Done, replaced with number config entity